lib/game.rb

```diff
@@ -0,0 +1,95 @@
+class Game
+
+  WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7],[2,5,8], [0,4,8], [6,4,2]]
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.any? do |arr|
+      if (board.cells[arr[0]] == "X" && board.cells[arr[1]] == "X" && board.cells[arr[2]] == "X") ||
+          (board.cells[arr[0]] == "O" && board.cells[arr[1]] == "O" && board.cells[arr[2]] == "O")
+          return arr
+      end
+    end
+  end
+
+  def draw?
+    !won? && board.full?
+  end
+
+  def winner
+    return board.cells[won?[0]] if won?
+  end
+
+  def turn
+    input = current_player.move(board)
+    if board.valid_move?(input)
+      board.update(input, current_player)
+      board.display
+    else
+      turn
+    end
+    current_player = current_player == player_1 ? player_2 : player_1
+  end
+
+  def play
+    board.display
+    until over?
+      turn
+    end
+    if won?
+      board.display
+      puts "Congratulations #{winner}!"
+    else
+      board.display
+      puts "Cat's Game!"
+    end
+  end
+
+  def self.start
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.